openvpn-wire-protocol.xml

```diff
@@ -1542,6 +1544,230 @@ struct datakeys {
+      <section title="Epoch key usage">
+        <t>
+          When sending packets, a peer will encrypt packet until the limit for the epoch K_i is reached. At this point,
+          the sender will switch to the next epoch key K_i+1. The state for the sending side, only needs to keep
+          the current E_i key and the current encryption key K_i.
+        </t>
+        <t>
+          Each peer is expected to hold the current epoch key K_i, the previous K_i-1 and a number of
+          future keys K_i+1, K_i+2,....,K_i+j. The number of future keys j depends on the expected possible
+          data rate of the implementation and needs to be large enough that even when a reasonable amount of
+          packets is lost, the implementation will still be able to authenticate the packets.
+        </t>
+
+        <t>
+          Allowing multiple epochs keys requires the receiving side to keep multiple keys to able to decrypt and
+          verify packets with different epochs. An attacker can send a packet with a epoch that is a lot higher than
+          the current epoch. In order to avoid needing to the generate/keep arbitrary epoch decryption keys, the window
+          of future decryption keys that is acceptable should be limited.  At the same time, the window needs to be large
+          enough that a long packet loss that causes all packets to be dropped (e.g. a topology change) does not
+          cause legitimate packets to be dropped.  
+        </t>
+        <t>
+          For example, if we consider an  implementation that allows to encrypt at data rate of 100 GBit/s and
+          to be able to deal with 60s of lost data, we have to calculate how many epochs the peer can use and
+          exhaust in this time. Assuming a small packet size of 128 byte, this gives us about 78 million packets
+          per second or about 2^26.2 packets per second. For the full minute we get 2^32.1 packets. Assuming
+          the limit of 2^28.3 packets per epoch for AES-GCM, we need 14 epoch keys to safely encrypt this
+          amount of data.  A reasonable amount of future epoch data channel keys in this scenario could be
+          considered 16 or 32 keys.
+        </t>
+        <t>
+          When receiving an authenticated packet with a higher epoch than the currently used one, the
+          receiver of this packet SHOULD also switch to the same epoch. This allow also the receiver to
+          trigger using a new encryption key if in the future it becomes necessary. But this is optional
+          and an implementation can continue using the old send key as log as it is safe. 
+        </t>
+        <t>
+          As addition a peer cannot regard apparent overuse of a send key by other the other peer as error as the
+          limit can change if new information is available but the peer MAY move to a higher sending key epoch of its
+          own to try to trigger the same on the peer as outlined in the previous paragraph.
```

Comment:
I'm not sure I understand this. In the future, there might be better bounds for GCM mode and we update our RFC to reflect them, and old implementations need to handle that situation?